uefi-test-runner/build.py

```diff
@@ -132,7 +130,29 @@ def run_qemu(headless):
     if VERBOSE:
         print(' '.join(cmd))
 
-    sp.run(cmd).check_returncode()
+    # This regex can be used to detect and strip ANSI escape codes when
+    # analyzing the output of the test runner.
+    ansi_escape = re.compile(r'(\x9B|\x1B\[)[0-?]*[ -/]*[@-~]')
+
+    # Start QEMU
+    qemu = sp.Popen(cmd, stdout=sp.PIPE, universal_newlines=True)
```

Comment:
We can keep using [subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run) here, just rename the `universal_newlines=True` parameter to `text=True`

Reply:
Use of popen is part of a longer-term evil plan to allow the host to take screenshots as QEMU is running. :smile: In the shorter time, it also allows the log lines to reach the host a bit faster (no need to wait for QEMU to close).

Use of `universal_newlines` is for broader Python compatibility. I did not want to depend on python 3.7 (which is not packaged yet on my Linux distro) just to shrink a function parameter.

Reply:
Ah, I didn't realize `universal_newlines` was added recently.